### PR TITLE
feat: add pagination and table components

### LINKED ISSUE #335: Add pagination
All index pages, apart from board page, need pagination.
You may use [flop](https://hexdocs.pm/flop/readme.html) to solve this.

### LINKED ISSUE #367: Add searching, sorting and filtering to memberships table
Maybe use [this](https://hexdocs.pm/flop_phoenix/readme.html) package.

### PR DESCRIPTION
**Pull request by joaodiaslobo**
Adds customizable table and pagination components. Uses [Flop Phoenix ](https://hexdocs.pm/flop_phoenix/0.14.2/Flop.Phoenix.html)for column filters inside the table. Fixes #367 , fixes #335 

image

### CODE REVIEW

**Review by MarioRodrigues10 — DISMISSED** [review]
Make sure every page is accessible, for example this one:
/organizations/:id returns an error when acessed
image

**Comment by MarioRodrigues10** [comment]
Also @joaodiaslobo and @RuiL1904 , do you think would make sense to have a role field in the following table:
image

File: lib/atomic_web/live/activity_live/index.ex [hunk 1]
```diff
@@ -75,13 +93,15 @@ defmodule AtomicWeb.ActivityLive.Index do
       )
   end
 
-  defp list_sessions(organization_id) do
-    Activities.list_sessions_by_organization_id(organization_id,
-      preloads: [:activity, :speakers, :enrollments]
-    )
-  end
+  defp list_sessions(id, params) do
```
Comment (ruilopesm) [inline lib/atomic_web/live/activity_live/index.ex]:
```suggestion
  defp list_sessions(id, params \\ %{}) do
```

File: lib/atomic_web/live/activity_live/index.ex [hunk 2]
```diff
@@ -67,6 +56,35 @@ defmodule AtomicWeb.ActivityLive.Index do
...
+  @impl true
+  def handle_event("delete", %{"id" => id}, socket) do
+    activity = Activities.get_activity!(id)
+    {:ok, _} = Activities.delete_activity(activity)
+
+    {:noreply, assign(socket, list_sessions(socket.assigns.current_organization.id, %{}))}
+  end
+
+  def handle_event("open-enrollments", _payload, socket) do
+    {:noreply, assign(socket, list_sessions(socket.assigns.current_organization.id, %{}))}
```
Comment (ruilopesm) [inline lib/atomic_web/live/activity_live/index.ex]:
```suggestion
    {:noreply, assign(socket, list_sessions(socket.assigns.current_organization.id))}
```

File: lib/atomic_web/live/activity_live/index.ex [hunk 3]
```diff
@@ -67,6 +56,35 @@ defmodule AtomicWeb.ActivityLive.Index do
     |> assign(:activity, nil)
   end
 
+  @impl true
+  def handle_event("delete", %{"id" => id}, socket) do
+    activity = Activities.get_activity!(id)
+    {:ok, _} = Activities.delete_activity(activity)
+
+    {:noreply, assign(socket, list_sessions(socket.assigns.current_organization.id, %{}))}
```
Comment (ruilopesm) [inline lib/atomic_web/live/activity_live/index.ex]:
```suggestion
    {:noreply, assign(socket, list_sessions(socket.assigns.current_organization.id))}
```

**Review by ruilopesm — COMMENTED** [review]
image
I'm getting this error while trying to access memberships page.

**Review by ruilopesm — DISMISSED** [review]
Excellent work! 🚀

**Comment by joaodiaslobo** [comment]
> Also @joaodiaslobo and @RuiL1904 , do you think would make sense to have a role field in the following table: image

That's certainly something that can be added, in this specific table the only memberships displayed are ones belonging to members, admins and owners. I'll add a new column for this field. 👍

**Review by ruilopesm — COMMENTED** [review]
I'm not being able to sort "Requested At" and "Last Update" fields.

**Review by ruilopesm — DISMISSED** [review]

**Review by ruilopesm — APPROVED** [review]

**Pull request merged**